docs/developer-guide/contributing.md

```diff
@@ -156,6 +156,29 @@ make: *** [Makefile:386: verify-kube-connect] Error 1
 
 you should edit your `~/.kube/config` and modify the `server` option to point to your correct K8s API (as described above).
 
+### If using k3d to spin-up your Kubernetes cluster
+
+Because it's running in a docker container, you're dealing with docker's internal networking rules when using k3d. This is the cost of a fully self-contained, disposable k8s cluster. The following steps should help with a successful `make verify-kube-connect` execution.
```

Comment:
Introduce k3d, provide some details about how it different than a usual kubernetes cluster - and the subsequent issues.

:)

Reply:
thanks for the note @sbose78 :)
Added an intro for k3d and some comparison with traditional k8s cluster on local machine.